src/contract.ts

```diff
@@ -1,376 +1,479 @@
-      programDerivedAddresses = [],
-      signers = [],
-      caller = this.program.payerAccount.publicKey,
-      value = 0,
-      simulate = false,
-    } = options ?? {};
-
-    const fragment = this.abi.getFunction(name);
-    const input = this.abi.encodeFunctionData(name, args);
-
-    const seeds = programDerivedAddresses.map((pda) => pda.seed);
-
-    const data = Buffer.concat([
-      this.storageAccount.toBuffer(), //                contract
-      caller.toBuffer(), //                             sender
-      Buffer.from(numToPaddedHex(value), 'hex'), //     value
-      Buffer.from('00000000', 'hex'), //                hash
-      encodeSeeds(seeds), //                                         seeds
-      Buffer.from(input.replace('0x', ''), 'hex'), //   input
-    ]);
-
-    const keys = [
-      ...programDerivedAddresses.map((pda) => ({
-        pubkey: pda.account,
-        isSigner: false,
-        isWritable: true,
-      })),
-      {
-        pubkey: this.storageAccount,
-        isSigner: false,
-        isWritable: true,
-      },
-      {
-        pubkey: SYSVAR_CLOCK_PUBKEY,
-        isSigner: false,
-        isWritable: false,
-      },
-      {
-        pubkey: PublicKey.default,
-        isSigner: false,
-        isWritable: false,
-      },
-      ...accounts.map((pubkey) => ({
-        pubkey,
-        isSigner: false,
-        isWritable: false,
-      })),
-      ...writableAccounts.map((pubkey) => ({
-        pubkey,
-        isSigner: false,
-        isWritable: true,
-      })),
-    ];
-
-    const instruction = new TransactionInstruction({
-      keys,
-      programId: this.getProgramKey(),
-      data,
-    });
-
-    signers.unshift(this.program.payerAccount);
-
-    const { encoded, logs, computeUnitsUsed } = await (simulate
-      ? this.program.simulateTransaction
-      : this.program.sendAndConfirmTransaction
-    ).call(this.program, [instruction], signers);
-
-    let result: Result | null = null;
-
-    if (fragment.outputs?.length) {
-      if (!encoded) {
-        throw new Error('return data not set');
-      }
-
-      result = this.abi.decodeFunctionResult(fragment, encoded);
-      if (fragment.outputs.length === 1) {
-        result = result[0];
-      }
+
+        const { logs, computeUnitsUsed } = simulate
+            ? await simulateTransactionWithLogs(this.connection, transaction, [payer, storage, ...signers])
+            : await sendAndConfirmTransactionWithLogs(this.connection, transaction, [payer, storage, ...signers]);
+
+        const events = this.parseLogsEvents(logs);
+
+        return {
+            logs,
+            events,
+            computeUnitsUsed,
+        };
+    }
+
+    /**
+     * Clone the contract. This creates a new contract with the same configuration but no log listeners.
+     *
+     * @return Clone of the contract
+     */
+    clone(): Contract {
+        return new Contract(this.connection, this.program, this.storage, this.abi, this.payer);
+    }
```

Comment:
This method is probably not very useful and could be trimmed. I just included it for something in the tests.